sdks/python/apache_beam/io/vcfio.py

```diff
@@ -0,0 +1,428 @@
+
+  A call represents the determination of genotype with respect to a particular
+  variant. It may include associated information such as quality and phasing.
+  """
+
+  def __init__(self, name=None, genotype=None, phaseset=None, info=None):
+    """Initialize the :class:`VariantCall` object.
+
+    Args:
+      name (str): The name of the call.
+      genotype (List[int]): The genotype of this variant call as specified by
+        the VCF schema. The values are either `0` representing the reference,
+        or a 1-based index into alternate bases. Ordering is only important if
+        `phaseset` is present. If a genotype is not called (that is, a `.` is
+        present in the GT string), -1 is used
+      phaseset (str): If this field is present, this variant call's genotype
+        ordering implies the phase of the bases and is consistent with any other
+        variant calls in the same reference sequence which have the same
+        phaseset value. If the genotype data was phased but no phase set was
+        specified, this field will be set to `*`.
+      info (dict): A map of additional variant call information. The key is
+        specified in the VCF record and the type of the value is specified by
+        the VCF header FORMAT.
+    """
+    self.name = name
+    self.genotype = genotype or []
+    self.phaseset = phaseset
+    self.info = info or {}
+
+  def __eq__(self, other):
+    return (self.name == other.name and
+            self.genotype == other.genotype and
+            self.phaseset == other.phaseset and
+            self.info == other.info)
+
+  def __repr__(self):
+    return ', '.join(
+        [str(s) for s in [self.name, self.genotype, self.phaseset, self.info]])
+
+
+class _VcfSource(filebasedsource.FileBasedSource):
+  """A source for reading VCF files.
+
+  Parses VCF files (version 4) using PyVCF library. If file_pattern specifies
+  multiple files, then the header from each file is used separately to parse
+  the content. However, the output will be a uniform PCollection of
+  :class:`Variant` objects.
+  """
+
+  DEFAULT_VCF_READ_BUFFER_SIZE = 65536  # 64kB
+
+  def __init__(self,
+               file_pattern,
+               compression_type=CompressionTypes.AUTO,
+               buffer_size=DEFAULT_VCF_READ_BUFFER_SIZE,
+               validate=True):
+    super(_VcfSource, self).__init__(file_pattern,
+                                     compression_type=compression_type,
+                                     validate=validate)
+
+    self._header_lines_per_file = {}
+    self._file_pattern = file_pattern
+    self._compression_type = compression_type
+    self._buffer_size = buffer_size
+    self._validate = validate
+
+  def read_records(self, file_name, range_tracker):
+    record_iterator = _VcfSource._VcfRecordIterator(
+        file_name,
+        range_tracker,
+        self._file_pattern,
+        0,  # min_bundle_size
+        self._compression_type,
```

Comment:
<!--new_thread; commit:63511ddb8caf9e5c12888ef8a4d1071e0ea6c168; resolved:0-->
Can we simply pass constant values specified here to TextSource and remove from the constructor of _VcfRecordIterator instead of passing here ?

Reply:
Great point, I'll change this.